schema/schema.go

```diff
@@ -35,13 +35,24 @@ var (
 	// having the OCI JSON schema files in root "/".
 	fs = _escFS(false)
 
-	// specs maps OCI schema media types to schema files.
+	// schemaNamespaces is a set of URI prefixes which contain the schema files of fs (used in JSON schema "id" and "$ref" attributes).
+	// fs stores all of its files in the root of the filesystem tree.
+	// (Note that this must contain subdirectories before its parent directories for fsLoaderFactory.refContents to work.)
```

Comment:
This needs stronger wording to indicate that these aren't actually hosted at the URIs provided and this is requirement of the fix to avoid remote IO.

Reply:
> This needs stronger wording to indicate that these aren't actually hosted at the URIs provided and this is requirement of the fix to avoid remote IO.

If there is no intention of hosting the files at the URIs listed in the `id` properties, why don't we just drop the `id` properties from the JSON Schema (like opencontainers/runtime-spec#945 is doing for runtime-spec)?  Then image-spec won't need to fake the presence of resources at those URIs with loader hoop-jumping and other JSON Schema consumers which ingest this JSON Schema without going through the gojsonschema loader will work too.

Reply:
Updated the comment to hopefully explain the situation, while still being concise enough.

I have left the more long-winded explanation of what the commit does, and why, and what it _doesn’t do_, to the commit message and the PR discussion. If you’d rather have more of the rationale in code comments, please say so.